api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java

```diff
@@ -105,39 +103,25 @@ public static ConfigProviderResolver instance() {
                 if (instance != null) {
                     return instance;
                 }
+                // first try my class loader
+                ConfigProviderResolver instance = loadSpi(ConfigProviderResolver.class.getClassLoader(), null);
+                // next try the system class loader
+                instance = loadSpi(null, instance);
```

Comment:
I'm really not sure if it makes sense to load the resolver by both the resolver's classloader and the system classloader. I'm afraid that if the resolver impl is in the system classloader and if resolver's classloader delegates to the system classloader, the same impl would be found in both classloaders and that would end with an exception "Multiple ConfigResolverProvider implementations found". Is it really necessary to try loading also with the system classloader? In which relevant case the resolver's classloader doesn't delegate to the system classloader so that the resolver impl isn't found and needs to be looked up by the system classloader? I would delete the call to loadSpi using the system classloader and only rely on the resolver's classloader.

Reply:
I would be happy to revise this patch to only consider the implementation's own class loader.  It would simplify the change.